src/SPC/builder/Extension.php

```diff
@@ -231,6 +254,32 @@ public function validate(): void
         // do nothing, just throw wrong usage exception if not valid
     }
 
+    public function buildShared(): void
+    {
+        match (PHP_OS_FAMILY) {
+            'Darwin', 'Linux' => $this->buildUnixShared(),
+            default => throw new WrongUsageException(PHP_OS_FAMILY . ' build shared extensions is not supported yet'),
+        };
+    }
+
+    public function buildUnixShared(): void
+    {
+        // prepare configure args
+        shell()->cd($this->source_dir)
+            ->setEnv(['CFLAGS' => $this->builder->arch_c_flags ?? ''])
+            ->execWithEnv(BUILD_BIN_PATH . '/phpize')
+            ->execWithEnv('./configure ' . $this->getUnixConfigureArg() . ' --with-php-config=' . BUILD_BIN_PATH . '/php-config --enable-static --disable-shared')
```

Comment:
--enable-static --disable-shared? We want --enable-shared and --disable-static for the shared extensions

Reply:
Makes sense.